scripts/build_ffmpeg.sh

```diff
@@ -0,0 +1,86 @@
+# !/bin/bash
+
+if [ "$#" -lt 2 ]; then
+	echo "usage build_ffmpeg <ffmpeg build path> <product path> <release/debug - optional>"
+	exit
+fi
+
+FFMPEG_BUILD_PATH=$1
+PRODUCT_ROOT_PATH=$2
+ADDON_BUILD_PATH=$PRODUCT_ROOT_PATH/node_addons/FormatConverter/build/
+FFMPEG_SYMLINK=$ADDON_BUILD_PATH/FFmpeg
+BUILD_CONF=
+OS=`uname`
+
+echo "PRODUCT_ROOT_PATH=$PRODUCT_ROOT_PATH"
+
+[ -d $FFMPEG_BUILD_PATH ] || mkdir -p "$FFMPEG_BUILD_PATH"
+
+[ "$#" -eq 3 ] && BUILD_CONF=$3 || BUILD_CONF=Release
+
+if [ "$BUILD_CONF" = "debug" ]; then
+	BUILD_CONF=Debug
+	echo "Debug mode"
+else
+	BUILD_CONF=Release
+	echo "Release mode"
+fi
+
+mkdir -p ${BUILD_CONF}
+
+
+echo "current path `pwd`"
+echo "FFMPEG_BUILD_PATH=$FFMPEG_BUILD_PATH"
+echo "ADDON_BUILD_PATH=$ADDON_BUILD_PATH"
+
+echo "unlink $FFMPEG_SYMLINK"
+unlink $FFMPEG_SYMLINK
+
+
+if [ ! -d $FFMPEG_SYMLINK ]; then
+
+    curl -OL https://github.com/FFmpeg/FFmpeg/releases/download/n3.0/ffmpeg-3.0.tar.gz
+    mv ./ffmpeg-3.0.tar.gz /var/tmp/ffmpeg-3.0.tar.gz
+
+	# note: if the second argument already exists and is a directory,
+	# ln will create a symlink to the target inside that directory.
+
+    tar -xzvf /var/tmp/ffmpeg-3.0.tar.gz -C $FFMPEG_BUILD_PATH
+    ln -s $FFMPEG_BUILD_PATH/ffmpeg-3.0 $FFMPEG_SYMLINK
+else
+	echo "$FFMPEG_SYMLINK exists skipping ffmpeg download"
+fi
+
+pushd $FFMPEG_SYMLINK
+
+    echo "current path `pwd`"
+
+    debug_specifics=
+    [ "$BUILD_CONF" = "Debug" ] &&  debug_specifics='--enable-debug --disable-optimizations'
+
+    configFileName=./lastConfigure
+
+    confCmd="./configure --disable-everything --disable-doc --enable-protocol=file --enable-demuxer=mpegts --enable-muxer=rtp_mpegts --enable-parser=h264 --enable-parser=aac --enable-muxer=mp4 --enable-zlib --enable-bsf=aac_adtstoasc --enable-decoder=aac --enable-decoder=h264 --enable-muxer=flv --enable-protocol=rtmp --enable-encoder=libmp3lame $debug_specifics"
+
+    [ $OS == "Linux" ] && confCmd="$confCmd --enable-pic"
```

Comment:
Why is this needed? ffmpeg correctly uses GNU's autotools to determine which build flags [CFLAGS and LDFLAGS]  are needed per OS and arch.